File: turbo/benchmarks/c/primes.c
#include <stdio.h>
#include <stdbool.h>

bool is_prime(long n) {
    if (n < 2) return false;
    for (long i = 2; i * i <= n; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int main() {
    int count = 0;
    for (long n = 2; n < 100000; n++) {
        if (is_prime(n)) count++;
    }
    printf("%d\n", count);
    return 0;
}

File: turbo/benchmarks/rust/primes.rs
fn is_prime(n: i64) -> bool {
    if n < 2 { return false; }
    let mut i = 2i64;
    while i * i <= n {
        if n % i == 0 { return false; }
        i += 1;
    }
    true
}
fn main() {
    let mut count = 0;
    for n in 2..100_000i64 {
        if is_prime(n) { count += 1; }
    }
    println!("{}", count);
}
